frontend/ui/network/manager.lua

```diff
@@ -105,23 +119,14 @@ function NetworkMgr:promptWifiOff(complete_callback)
 end
 
 function NetworkMgr:turnOnWifiAndWaitForConnection(callback)
-    NetworkMgr:turnOnWifi()
-    local timeout = 30
-    local retry_count = 0
-    local info = InfoMessage:new{ text = T(_("Enabling Wi-Fi. Waiting for Internet connection…\nTimeout %1 seconds."), timeout)}
+    local info = InfoMessage:new{ text = _("Enabling Wi-Fi. Waiting for Internet connection…") }
     UIManager:show(info)
     UIManager:forceRePaint()
-    while not NetworkMgr:isOnline() and retry_count < timeout do
```

Comment:
Yes, this is swapped to an `isConnected` on purpose.

It's generally good enough, and if WAN connectivity happens to be required by the callback, it'll just timeout on its own.